plugins/providers/gcloudiam/provider.go

```diff
@@ -50,6 +52,41 @@ func (p *Provider) GetResources(pc *domain.ProviderConfig) ([]*domain.Resource,
 		t = ResourceTypeOrganization
 	}
 
+	client, err := p.getIamClient(pc)
+	if err != nil {
+		return nil, err
+	}
+
+	iamRoles, err := client.GetRoles()
+	if err != nil {
+		return nil, err
+	}
+
+	var roles []*domain.Role
+	for _, r := range iamRoles {
+		roles = append(roles, &domain.Role{
+			ID:          r.Name,
+			Name:        r.Title,
+			Description: r.Description,
+		})
+	}
+
+	rolesMap := make(map[string]*domain.Role)
+	for _, role := range roles {
+		rolesMap[role.ID] = role
+	}
+
+	for _, rc := range pc.Resources {
+		for _, ro := range rc.Roles {
+			for _, p := range ro.Permissions {
+				permission := fmt.Sprint(p)
+				if _, ok := rolesMap[permission]; !ok {
+					return nil, fmt.Errorf("%v: %v", ErrInvalidProjectRole, permission)
+				}
+			}
+		}
+	}
+
```

Comment:
this permission validation should be part of `CreateConfig` instead of `GetResources` so that when a user creates/updates a provider config with invalid permissions they will got `ErrInvalidProjectRole`/`ErrInvalidPermission` immediately.

Reply:
Done